### PR TITLE
Suggest how to handle multiple trackers.

### PR DESCRIPTION
**Pull request by Karpo72**
Array 'trackingSupportingMultipleTrackers' defines analytics commands that support multiple trackers.
Multiple trackers are then supported at one single point in the code.

TODO: Removing existings loops for multiple trackers

### CODE REVIEW

File: src/angular-google-analytics.js [hunk 1]
```diff
@@ -187,13 +192,29 @@
         };
 
         var _ga = function() {
+          var tmpCommand, tmpArguments;
```
Comment (justinsa) [inline src/angular-google-analytics.js]: You don't need these defined in this scope as they are only used within the anonymous function passed into the forEach on line 208.

File: src/angular-google-analytics.js [hunk 2]
```diff
@@ -187,13 +192,29 @@
...
+          // Temporary check if command contains '.' until all functions are switched
+          if (angular.isArray(accountId) && arguments && arguments.length > 1 && arguments[0].indexOf('.') < 0 &&
+              trackingSupportingMultipleTrackers.indexOf(arguments[0]) >= 0) {
+            // Create command for each tracker an apply
+            tmpArguments = arguments
+            accountId.forEach(function (trackerObj) {
+              tmpCommand = tmpArguments[0];
+              tmpCommand = trackerObj.name + '.' +tmpCommand;
+              tmpArguments[0] = tmpCommand;
+              $window.ga.apply(null, tmpArguments);
```
Comment (justinsa) [inline src/angular-google-analytics.js]: This implementation is going to have problems due to tmpArguments being set to arguments on line 207. On line 211 the 0 index then gets overwritten and that will be used on the following iteration. The tests should have caught this not working correctly.

The following will work correctly and removes the need for line 195 and 207 as well.
```
var command = trackerObj.name + '.' + arguments[0],
      args = arguments.slice();
args[0] = command;
$window.ga.apply(null, args);
```

File: src/angular-google-analytics.js [hunk 3]
```diff
@@ -23,7 +23,12 @@
           linkerConfig = {'allowLinker': true},
           trackUrlParams = false,
           delayScriptTag = false,
-          logAllCalls = false;
+          logAllCalls = false,
+          trackingSupportingMultipleTrackers = [
+            'ec',
+            'ecommerce',
+            'send'
+          ];;
```
Comment (justinsa) [inline src/angular-google-analytics.js]: Two semicolons. The linter should have caught this.

File: src/angular-google-analytics.js (hunk 2 above)
Comment (Karpo72) [inline src/angular-google-analytics.js]: ok...take a look....I changed the code a little bit

Comment (Karpo72) [inline src/angular-google-analytics.js]:
```javascript
if (angular.isArray(accountId) && arguments && arguments.length > 1 && arguments[0].indexOf('.') < 0 &&
              trackingSupportingMultipleTrackers.indexOf(arguments[0]) >= 0) {
            var args = Array.prototype.slice.call(arguments);
            // Create command for each tracker an apply
            accountId.forEach(function (trackerObj) {
              var command = trackerObj.name + '.' + args[0];
              args[0] = command;
              $window.ga.apply(null, args);

            });
          } else {
            $window.ga.apply(null, arguments);
          }
```

File: src/angular-google-analytics.js [hunk 4]
```diff
@@ -187,13 +192,28 @@
...
+          // If multiple trackers && arguments valid && command supports multiple trackers
+          // Temporary check if command contains '.' until all functions are switched
+          if (angular.isArray(accountId) && arguments && arguments.length > 1 && arguments[0].indexOf('.') < 0 &&
+              trackingSupportingMultipleTrackers.indexOf(arguments[0]) >= 0) {
+            var args = Array.prototype.slice.call(arguments);
+            // Create command for each tracker an apply
+            accountId.forEach(function (trackerObj) {
+              var command = trackerObj.name + '.' + args[0];
+              args[0] = command;
+              $window.ga.apply(null, args);
```
Comment (justinsa) [inline src/angular-google-analytics.js]: This is still problematic and won't work properly. The code needs to be:

```javascript
var command = trackerObj.name + '.' + arguments[0],
      args = arguments.slice();
args[0] = command;
$window.ga.apply(null, args);
```

The justification is as follows. Setting args on line 206 to a copy of the original arguments does not allow it to be reset on each loop of the ```forEach``` on line 208. The consequence is that when ```args[0]``` is set on line 210 that new value will then be used on line 209 on each subsequent iteration of the ```forEach```. As an illustrative example:

Let, ```args = ['x', 'y', 2]``` and ```accountId = [{ name: 'name1' }, { name: 'name2' }]```

On Iteration #1 we will get:
   trackerObj = { name: 'name1' }
   command = 'name1.x'
   args = ['name1.x', 'y', 2]

On iteration #2 we will get:
   trackerObj = { name: 'name2' }
   command = 'name2.name1.x'
   args = ['name2.name1.x', 'y', 2]

I hope that makes it clear what is wrong with the implementation in this commit. Additionally, if the unit tests aren't catching this then a new unit test should be added for ensuring existing behaviors are preserved and the expected behavior of this function is maintained.

Comment (justinsa) [inline src/angular-google-analytics.js]: Alternatively, though it is a micro-optimization, you could add a statement at line 209 to set ```args[0] = arguments[0]``` so the first index is reset in the args array on each iteration.

File: src/angular-google-analytics.js [hunk 5]
```diff
@@ -187,13 +192,28 @@
...
+
```
Comment (justinsa) [inline src/angular-google-analytics.js]: Please remove this empty line.

File: src/angular-google-analytics.js [hunk 6]
```diff
@@ -187,13 +192,28 @@
         };
 
         var _ga = function() {
+
```
Comment (justinsa) [inline src/angular-google-analytics.js]: Please remove this empty line.

**Comment by justinsa** [comment]
@Karpo72 I realize you are calling this PR a suggestion and I am treating it like a PR you wanted pulled in, which may not be your intention so I figured I should confirm with you. Are you going to handle the TODO you added and any changes to test cases needed or would you prefer I take ownership of that? If you want me to take ownership of that then I will adopt your changes in a separate PR that will include the necessary additional changes for the TODO and unit tests. Thanks.

**Pull request closed without merging**